src/azurelinux/3.0/cross/amd64-ubuntu-16.04/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-crossdeps-builder-local AS builder
+ARG ROOTFS_DIR=/crossrootfs/x64
+
+RUN /scripts/eng/common/cross/build-rootfs.sh x64 xenial --skipunmount
+
+# Build compiler-rt profile library for PGO instrumentation
+RUN mkdir compiler-rt_build && cd compiler-rt_build && \
+    BUILD_FLAGS="-v --sysroot=$ROOTFS_DIR" \
+    TARGET_TRIPLE="x86_64-linux-gnu" && \
+    cmake ../llvm-project.src/compiler-rt \
+        -DCOMPILER_RT_BUILD_PROFILE=ON \
+        -DCOMPILER_RT_BUILD_SANITIZERS=ON \
+        -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON \
+        \
+        -DCOMPILER_RT_BUILD_BUILTINS=OFF \
+        -DCOMPILER_RT_BUILD_CRT=OFF \
+        -DCOMPILER_RT_BUILD_GWP_ASAN=OFF \
+        -DCOMPILER_RT_BUILD_LIBFUZZER=OFF \
+        -DCOMPILER_RT_BUILD_MEMPROF=OFF \
+        -DCOMPILER_RT_BUILD_ORC=OFF \
+        -DCOMPILER_RT_BUILD_XRAY=OFF \
+        \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_ASM_COMPILER=clang \
+        -DCMAKE_C_COMPILER=clang \
+        -DCMAKE_CXX_COMPILER=clang++ \
+        -DCMAKE_SHARED_LINKER_FLAGS="-fuse-ld=lld" \
+        -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=lld" \
+        -DCMAKE_ASM_COMPILER_TARGET="${TARGET_TRIPLE}" \
+        -DCMAKE_C_COMPILER_TARGET="${TARGET_TRIPLE}" \
+        -DCMAKE_CXX_COMPILER_TARGET="${TARGET_TRIPLE}" \
+        -DCMAKE_ASM_FLAGS="${BUILD_FLAGS}" \
+        -DCMAKE_C_FLAGS="${BUILD_FLAGS}" \
+        -DCMAKE_CXX_FLAGS="${BUILD_FLAGS}" && \
+    make -j $(getconf _NPROCESSORS_ONLN)
+
+RUN LLVM_VERSION=16.0.0 LLVM_VERSION_MAJOR="${LLVM_VERSION%%.*}" SANITIZER_RUNTIMES_DIR=/sanitizer-runtimes && \
```

Comment:
LLVM 16 makes sense for .NET 8 since that's what we started with.

18 is already out. We should decide our approach for .NET 9+. Should we definite a peer set of these images that includes LLVM 18? I'm guessing by the time .NET 10 comes around that we'll be looking at LLVM 20.

Reply:
I'd prefer -llvmXX over -netXX as it's more descriptive of what's contained in the package.

Reply:
> I'd prefer -llvmXX over -netXX as it's more descriptive of what's contained in the package.

It is impossible to fully describe what's in the image in the name. I expect that we would need the keep tweaking the naming scheme as new requirements show up.

`-netXX` naming is more durable. It simply means "image used to build .NET XX release" whatever that means.

Reply:
The `-netXX` scheme is fine with me.

Reply:
The advantage of that scheme is that it gives people a hint that a `net80` image probably should NOT still be in use in a .NET 9 or 10 build. Or ... I'm hopeful of that.